src/dmd/mars.d

```diff
@@ -1566,6 +1582,21 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
             params.map = true;
         else if (arg == "-multiobj")
             params.multiobj = true;
+        else if (startsWith(p + 1, "mixin="))
+        {
+            if (params.mixinFile)
+            {
+                error("-mixin=<file> can only be provided once!");
+                break;
+            }
+            auto tmp = p + 6 + 1;
+            if (!tmp[0])
+                goto Lnoarg;
+            //Don't rely on argv being valid after main has returned, this is used in atexit.
+            params.mixinFile = mem.xstrdup(tmp);
+            params.mixinOut = new OutBuffer;
+            atexit(&flushMixins);
```

Comment:
Not too happy about the use of atexit, but I agree that the result might be of interest even in the case of errors. That makes it a bit difficult to place it somewhere in tryMain in mars.d. Maybe add it as a scope(exit) flushMixins() there?

Reply:
Does that catch segfautls (e.g. Windows/OSX)?

Reply:
You mean if `flushMixins` when the compiler segfaults? Functions registered using `ataxia` are not called if the application terminates due to a segfault on macOS.

Reply:
Well, `scope(exit)` it is then.